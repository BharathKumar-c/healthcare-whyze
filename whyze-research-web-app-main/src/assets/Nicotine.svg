<svg width="40" height="40" viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg">
<rect width="40" height="40" rx="20" fill="#E9FCFC"/>
<path d="M13.2877 25.6633C13.8877 26.2633 14.6752 26.5633 15.5002 26.5633C16.3252 26.5633 17.1127 26.2258 17.7127 25.6633L29.9752 13.4008C30.3127 13.0633 30.5002 12.6133 30.5002 12.1633C30.5002 11.7133 30.3127 11.2633 29.9752 10.9258L28.0627 9.01328C27.7252 8.67578 27.2752 8.48828 26.8252 8.48828C26.3752 8.48828 25.9252 8.67578 25.5877 9.01328L13.2877 21.2758C12.6877 21.8758 12.3877 22.6633 12.3877 23.4883C12.3877 24.3133 12.6877 25.1008 13.2877 25.6633ZM26.4877 9.91328C26.5627 9.83828 26.6752 9.76328 26.8252 9.76328C26.9377 9.76328 27.0502 9.80078 27.1627 9.91328L29.0752 11.8258C29.1502 11.9008 29.2252 12.0133 29.2252 12.1633C29.2252 12.2758 29.1877 12.3883 29.0752 12.5008L19.5877 21.9133L17.0377 19.3633L26.4877 9.91328ZM14.2252 22.2133L16.1377 20.3008L18.6877 22.8508L16.7752 24.7633C16.4377 25.1008 15.9877 25.2883 15.5002 25.2883C15.0127 25.2883 14.5627 25.1008 14.2252 24.7633C13.8877 24.4258 13.7002 23.9758 13.7002 23.4883C13.7002 23.0008 13.8502 22.5508 14.2252 22.2133Z" fill="#0D5959"/>
</svg>
