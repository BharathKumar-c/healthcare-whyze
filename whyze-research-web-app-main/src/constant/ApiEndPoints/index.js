const FEASIBILITY_API = process.env.REACT_APP_FEASIBILITY_STUDY_BASE_URL;
const AUTH_API = process.env.REACT_APP_AUTH_URL;

export const feasibilityStudyApiUrls = {
  medicalConditionSearchUrl: `${FEASIBILITY_API}/search/indication`,
  medicationConditionSearchUrl: `${FEASIBILITY_API}/search/medication`,
  caseStudyUrl: `${FEASIBILITY_API}/case`,
  locationUrl: `${FEASIBILITY_API}/case/location`,
  initialPopulation: `${FEASIBILITY_API}/case/get-count`,
  allergiesListSearchUrl: `${FEASIBILITY_API}/search/allergies`,
  reactionListSearchUrl: `${FEASIBILITY_API}/search/reaction`,
  sitesDataUrl: `${FEASIBILITY_API}/case/site`,
  sitesWithPatientsCountUrl: `${FEASIBILITY_API}/case/site-patient-count`,
};

export const authEndPoints = {
  login: `${AUTH_API}/login`,
  register: `${AUTH_API}/register`,
  refreshToken: `${AUTH_API}/refreshToken`,
  forgotPassword: `${AUTH_API}/sendResetPasswordLink`,
  resetPassword: `${AUTH_API}/resetPassword`,
};

export const dashboardApiUrls = {
  projectUrl: `${FEASIBILITY_API}/project`,
  favouriteUrl: `${FEASIBILITY_API}/project/favourite`,
};

export const profileApiUrls = {
  image: `${AUTH_API}/image`,
  user: `${AUTH_API}/user`,
  profilePicture: `${AUTH_API}/profilePicture`,
};
