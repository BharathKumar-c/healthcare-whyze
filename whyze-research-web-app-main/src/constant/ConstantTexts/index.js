import React from 'react';

// dashboard welcome Screen const text

export const welComeComponentContent =
  'See how Whyze is preforming for researchers like you.';
export const createFirstStudy = 'Create your first study';
export const createFirstStudyContent =
  'Define your protocol or create your first feasibility study';
export const patientstitleLable = 'Patients';
export const totalPatientsLable = 'Total patients';
export const countriesLable = 'Countries';
export const sitesLable = 'Sites';
export const sitesRecruitedLable = 'Sites recruited';
export const avgTimeLable = 'Avg. response time';
export const trialsLable = 'Trials';
export const activeTrials = 'Active trials';
export const totalTrials = 'Total trials';
export const screenFailRate = 'Screen fail rate';
export const threeMonths = '3 months';
export const avgTimeToRecruit = 'Avg. time to recruit';
export const feasibilityStudiesInPlanning = 'Feasibility studies in planning';
export const seeAll = 'See all';

//Profile Page const text
export const ResearcherTitle = 'Researcher title -';
export const AuthProfile = 'AUTHENTICATED PROFILE';
export const ContactDetails = 'Contact details';
export const CompanyEmail = 'Company email address';
export const PhoneNumber = 'Phone';

//ContactModal Page Const texts

export const EmailAddress = 'Email address';
export const Phone = 'Phone';
export const okButtonLabel = 'Ok';
export const saveButtonLabel = 'save';
export const cancelButtonLabel = 'cancel';
export const contactDetailsLabel = 'Contact details';

//Addcriteria texts
export const addCriteriaTitle = 'Select a criteria item';

//RenameModal texts
export const renameModalTitle = 'Profile details';
export const fullNameLabel = 'Full name';
export const titleLabel = 'Title';

// Medication constant texts
export const medicationConstants = {
  medicationLabel: 'Medication',
  medicationErrorMessage: 'Please add the Medication',
  selectlabel: 'Select Dosage',
  inputLabel: 'Enter value',
  startedPrefixText: 'Started at least',
};
// Indication constant texts
export const indicationConstants = {
  medicalConditionLabel: 'Medical condition',
  medicalConditionErrorMessage: 'Please add the Medical condition',
};

//Allergies constant texte
export const allergiesConstants = {
  allergiesLabel: 'Allergies criteria',
  allergiesPlaceholder: 'Allergies',
  allergiesErrorText: 'Please add the Allergies',
};
//Symptom constant texts

export const symptomConstats = {
  symptomLabel: 'Symptom criteria',
  symptomPlaceholder: 'Symptom',
};
//Project settingmodal Constants

export const projectSettingConstants = {
  addProjectError: 'Please add the project',
  addProjectLabel: 'Project',
  addClientNameError: 'Please add ClientName',
  rangeLabel: 'Select',
  addClientLabel: 'Client name',
  description: 'Description',
  sharingLabel: 'Sharing',
  shareBtnLabel: 'Share',
  standardsText: 'Standards',
  standardDescription: 'Standardising markers and value outputs',
  measurementText: 'Default measurement',
  clinicalTrialError: 'Please add the clinical trial',
  clinicalTrial: 'Clinical Trial',
  sponsorError: 'Please add the sponsor',
  sponsor: 'Sponsor',
  projectSettings: 'Project settings',
  descriptionError: 'Please add the description',
  standardsError: 'Please select the standards',
  noIndication: 'No Indication',
};
export const shareProjectConstants = {
  addPeoplePlaceHolder: 'Add people and groups',
  accessPeopleText: 'People with access',
};
export const selectedPeoplesConstants = {
  addPeople: 'Add people and groups',
  notifyPeopleLabel: 'Notify people',
  messageLabel: 'Message',
};
// Nicotine constant texts
export const nicotineConstants = {
  nicotineConditionErrorMessage: 'Please add the nicotine condition',
  nicotineLabel: 'Nicotine criteria',
};

// Create Account constant texts
export const createAccountContants = {
  createAccounTitle: (
    <>
      Set up your <br /> Whyze account
    </>
  ),
  setUpWhyzeAccount: 'Set up your Whyze account',
  nameLabel: 'Name',
  emailLabel: 'Company email',
  phoneLabel: 'Your phone number',
  passwordLabel: 'Create password',
  characterLengthLabel: 'Between 8 and 20 characters',
  numericLabel: 'At least one number',
  uppercaseLabel: 'At least one upper-case letter',
  lowercaseLabel: 'At least one lower-case letter',
  createAccountButtonLabel: 'Create account',
  agreeConditionText: 'By clicking the button above you agree to our',
  termsOfUseText: 'Terms of use',
  privacyNoticesText: 'Privacy policies',
  alreadyHaveAccountText: 'Already have an account?',
  loginText: 'Login',
  termsOfUseLink: 'https://www.whyzehealth.com/privacy-notice/',
  privacyNoticesLink: 'https://www.whyzehealth.com/terms',
};

//indication const

export const conditions = {
  started: 'started',
  dosage: 'dosage',
  courseOfTherapy: 'course of therapy',
  diagnosed: 'diagnosed',
  recovered: 'recovered',
  severity: 'severity',
};

//demographics const

export const demographicsConst = {
  age: 'age',
  min: 'min',
  max: 'max',
  gender: 'gender',
};

//sites const

export const siteConst = {
  siteInformationCosnt: {
    patientSegmentHeader: 'Patient segment',
    connectedWithSite: 'Connected with this site',
    additionalCatchmentAreas:
      'Additional Whyze patients in site catchment areas',
    sitePerformance: 'Site performance',
    recruitmentTimeline: 'Recruitment timeline',
    dropOutRate: 'Drop out rate',
    trackRecordOfDoingTrials: 'Track record of doing trials',
    completedInThePastyear: 'Completed in the past year',
    primaryInvestigator: 'Primary Investigator',
    administration: 'Administration',
    screenFailRate: 'Screen fail rate',
    industryAverage: 'Industry average',
    avgResponseTime: 'Avg. response time',
  },
};

//setupAccount const

export const setupAccountConst = {
  dontHaveaccount: 'Don’t have an account yet?',
  setUpAccount: 'Set up account',
};

//forgotpassword const

export const forgotpasswordConst = {
  WHYZE: 'WHYZE',
  forgotPassword: 'Forgot Password',
  resendEmail: 'Enter your email below and we will send you a reset email.',
  pleaseTry: 'Didn’t receive it ? Please try again',
  returnTo: 'return to',
  loginPage: 'login',
};

//weight const  texts

export const weightConst = {
  weightRange: 'Weight Range',
  weightMin: 'min',
  weightMax: 'max',
  weightType: 'number',
  weightmaxErrorText: 'add max weight',
  weightminErrorText: 'add min weight',
  weightFrom: 'From',
  weightTo: 'To',
};

export const SaveAndCloseModalConst = {
  SaveAndCloseModalTitle: 'Name before exiting',
  projectNameInputLabel: 'Project',
  projectNameInputPlaceholder: 'New patient segment',
  clientNameInputLabel: 'Client name',
  descriptionNameInputLabel: 'Description',
  patientSegmentInputLabel: 'Patient segment required',
};

// Proposed Site list constant texts

export const proposedSiteListConstants = {
  title: 'Proposed Site List',
  seeMore: 'See more',
  emptyContent:
    'To create your proposed list, drag & drop or add your selected sites here.',
};

// Surgery constant text
export const surgeryConstants = {
  surgeryInputLabel: 'Surgery criteria',
  surgeryInputErrorMessage: 'Please add the Surgery',
};

export const feasibilityStudy = {
  newFeasibilityStudy: 'new-feasibility-study',
};
