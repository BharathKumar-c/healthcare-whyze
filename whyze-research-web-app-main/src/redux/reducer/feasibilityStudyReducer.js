import { createSlice } from '@reduxjs/toolkit';
import { v4 as uuid } from 'uuid';

export const initialState = {
  segments: [
    {
      name: 'Segment-1',
      isDone: false,
      activeTab: {
        name: 'feasibility',
        index: 0,
      },
      tabs: [
        {
          name: 'feasibility',
          title: 'Feasibility',
          isActive: true,
          position: 'left',
        },
      ],
      feasibilityStudy: [
        {
          name: 'Indication',
          case: 'Indication',
          condition: 'Inclusion',
          caseName: '',
          child: [],
        },
        {
          name: 'Demographics',
          case: 'Demographics',
          condition: 'Inclusion',
          caseName: '',
          child: [
            {
              id: uuid().slice(0, 8),
              condition: 'Gender',
              field: 'gender',
              gender: '',
            },
            {
              id: uuid().slice(0, 8),
              condition: 'Age range',
              field: 'age',
              ageRange: {
                min: 0,
                max: 0,
              },
            },
          ],
        },
      ],
      country: [],
      location: [],
      selectedCountries: [],
      totalConnectedNotConnected: {},
      sites: [],
      sitesContacted: [],
      sitesInitiated: [],
      preferredSites: [],
      patientCount: {
        feasibility: 0,
        location: 0,
      },
      siteCount: {
        feasibility: 0,
        location: 0,
      },
    },
  ],
  cardValue: {
    case: 'Indication',
    condition: '',
    index: 0,
  },
  activeSegment: {
    name: '',
    index: 0,
  },
};

const feasibilityStudySlice = createSlice({
  name: 'feasibilityStudyAction',
  initialState: initialState,
  reducers: {
    setCardDetails: (state, { payload }) => {
      state.cardValue = payload;
    },
    setSegments: (state, { payload }) => {
      state.segments = payload;
    },
    setActiveSegment: (state, { payload }) => {
      state.activeSegment = payload;
    },
    resetSegments: (state) => {
      state.segments = initialState.segments;
      state.cardValue = {
        case: 'Indication',
        condition: '',
        index: 0,
      };
      state.activeSegment = {
        name: '',
        index: 0,
      };
    },
  },
});

export const { setCardDetails, setSegments, setActiveSegment, resetSegments } =
  feasibilityStudySlice.actions;

export default feasibilityStudySlice.reducer;
