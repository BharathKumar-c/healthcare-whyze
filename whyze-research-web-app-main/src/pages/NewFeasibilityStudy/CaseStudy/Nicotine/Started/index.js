import React from 'react';
import PropTypes from 'prop-types';
import { Row, Space, Tag } from 'antd';
import { Image, Input } from '../../../../../components/basic';
import { ConditionContainer } from '../../../../../components/shared';
import DragIcon from '../../../../../assets/DragIcon.svg';
import CloseIcon from '../../../../../assets/CloseIcon.svg';
import { Vector } from '../../../../../assets';
import { useSelector, useDispatch } from 'react-redux';
import { removeChildInCaseStudy } from '../../../../../components/shared/AddCondition';
import { setSegments } from '../../../../../redux/reducer/feasibilityStudyReducer';
import { totalPatientFormatter } from '../../../../../utils/calculations';

export default function Started({
  totalpatient,
  parentIndex,
  caseDetails,
  isChild,
  parentId,
}) {
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const dispatch = useDispatch();

  const handleRemoveCondition = (id) => {
    const study = JSON.parse(JSON.stringify(segments));
    removeChildInCaseStudy(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      parentId,
      id,
    );
    dispatch(setSegments(study));
  };

  const handleInputChange = (event) => {
    const study = JSON.parse(JSON.stringify(segments));
    study[activeSegment.index].feasibilityStudy[parentIndex].child[0].started[
      event.target.name
    ] = event.target.value;
    dispatch(setSegments(study));
  };

  return (
    <ConditionContainer
      className={'condition-container'}
      isActive
      isVertical
      isChild={isChild}
    >
      <Space direction="vertical" className="container">
        <div className="started-container-alignment">
          <div className="drag-drop-image">
            <Image
              alt={'drag-icon'}
              src={DragIcon}
              width={10}
              height={16}
              preview={false}
            />
          </div>
          <div className="condition-container-input">
            <Input
              type="number"
              lable="From"
              name="start"
              value={caseDetails?.started?.start}
              handleInputChange={handleInputChange}
              onWheel={(e) => e.target.blur()}
            />
          </div>
          <div className="condition-container-input">
            <Input
              type="number"
              lable="To"
              name="to"
              value={caseDetails?.started?.to}
              handleInputChange={handleInputChange}
              onWheel={(e) => e.target.blur()}
            />
          </div>
          <div className="close-icon">
            <Image
              alt={'close-icon'}
              src={CloseIcon}
              className="image"
              width={20}
              height={20}
              preview={false}
              onClick={() => handleRemoveCondition(caseDetails.id, parentId)}
            />
          </div>
        </div>
        <Row className="indication-no-patient-tag-row">
          <Tag className="indication-no-patient-tag">
            <Vector />
            {totalPatientFormatter(totalpatient || 0)}
          </Tag>
        </Row>
      </Space>
    </ConditionContainer>
  );
}

Started.propTypes = {
  parentIndex: PropTypes.number,
  isChild: PropTypes.bool,
  totalpatient: PropTypes.string,
  parentId: PropTypes.string,
  caseDetails: PropTypes.object,
};

Started.defaultProps = {
  parentIndex: 0,
  isChild: false,
  totalpatient: '',
  parentId: '',
  caseDetails: null,
};
