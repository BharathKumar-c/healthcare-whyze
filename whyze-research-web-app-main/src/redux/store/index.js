import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
