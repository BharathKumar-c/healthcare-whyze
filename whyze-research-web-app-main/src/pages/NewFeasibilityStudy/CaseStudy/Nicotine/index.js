import React, { useState } from 'react';
import { Image, Select } from '../../../../components/basic';
import AddIcon from '../../../../assets/AddIcon.svg';
import PropTypes from 'prop-types';
import Quit from './Quit';
import Started from './Started';
import { nicotineSearchBoxOption } from '../../../../constant';
import { nicotineConstants } from '../../../../constant/ConstantTexts';
import { useSelector } from 'react-redux';
import { useDispatch } from 'react-redux';
import { setSegments } from '../../../../redux/reducer/feasibilityStudyReducer';
import { v4 as uuid } from 'uuid';

const Nicotine = (props) => {
  const [isError, setIsError] = useState(false);
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );

  const dispatch = useDispatch();

  const handleChangeCaseName = (e) => {
    const updatedSegments = JSON.parse(JSON.stringify(segments));
    const currentSegment = updatedSegments[activeSegment.index];
    const feasibilityStudy = currentSegment.feasibilityStudy[props.parentIndex];
    feasibilityStudy.caseName = e;

    if (e === 'no') {
      feasibilityStudy.child.pop();
    }

    if (feasibilityStudy.child.length > 0) {
      const condition = e === 'yes' ? 'started' : 'quit';
      const object =
        condition === 'started'
          ? { start: '', to: '' }
          : { startYear: '', quitYear: '' };

      delete feasibilityStudy.child[0][feasibilityStudy.child[0].condition];
      feasibilityStudy.child[0].condition = condition;
      feasibilityStudy.child[0] = {
        ...feasibilityStudy.child[0],
        [condition]: object,
      };
    }

    dispatch(setSegments(updatedSegments));
    setIsError(false);
  };

  const handleAddCondition = (index) => {
    const caseName = props?.caseDetails?.caseName;
    const condition = caseName === 'yes' ? 'started' : 'quit';
    const object =
      condition === 'started'
        ? { start: '', to: '' }
        : { startYear: '', quitYear: '' };
    const updatedSegments = JSON.parse(JSON.stringify(segments));
    const feasibilityStudy =
      updatedSegments[activeSegment.index].feasibilityStudy[index];
    const newChild = { id: uuid().slice(0, 8), condition, [condition]: object };
    feasibilityStudy.child = [...feasibilityStudy.child, newChild];
    dispatch(setSegments(updatedSegments));
  };

  return (
    <div className="indication" id="indication">
      {!props.isChild && (
        <>
          <Select
            id="main_textbox"
            className="nicotine-search-box"
            lable={nicotineConstants.nicotineLabel}
            getPopupContainer={() => document.getElementById('indication')}
            options={nicotineSearchBoxOption}
            handleSelect={handleChangeCaseName}
            value={props?.caseDetails?.caseName}
            isError={isError}
            errorText={nicotineConstants.nicotineConditionErrorMessage}
          />
          {props?.caseDetails?.caseName !== 'no' &&
            props?.caseDetails?.child.length < 1 && (
              <div style={{ height: '30px' }}>
                <Image
                  alt="nicotine-add-icon"
                  style={{ cursor: 'pointer' }}
                  preview={false}
                  width={12}
                  height={12}
                  src={AddIcon}
                  onClick={() =>
                    props.caseDetails.caseName !== ''
                      ? handleAddCondition(props.parentIndex)
                      : setIsError(true)
                  }
                />
              </div>
            )}
        </>
      )}
      {props.caseDetails?.child?.map((c, index) => {
        return (
          <div key={`${index}_${c.condition}_${c.operation}`}>
            {props.isChild ? (
              <div
                className={
                  c?.condition === ''
                    ? 'vertical-line-condition'
                    : 'vertical-line'
                }
              />
            ) : (
              <div
                data-testid="vertical-line"
                style={{
                  borderLeft: `2px ${
                    c?.condition === '' ? 'dashed' : 'solid'
                  } #D3D6D9`,
                  height: '32px',
                }}
              />
            )}
            {c?.condition === 'quit' &&
              props?.caseDetails?.caseName === 'notAnyMore' && (
                <Quit
                  caseDetails={c}
                  isChild={props.isChild}
                  parentIndex={props.parentIndex}
                  parentId={props.caseDetails.id}
                  totalpatient={c?.count}
                />
              )}
            {c.condition === 'started' &&
              props?.caseDetails?.caseName === 'yes' && (
                <Started
                  caseDetails={c}
                  isChild={props.isChild}
                  parentIndex={props.parentIndex}
                  parentId={props.caseDetails.id}
                  totalpatient={c?.count}
                />
              )}
          </div>
        );
      })}
    </div>
  );
};

Nicotine.propTypes = {
  caseDetails: PropTypes.object,
  parentIndex: PropTypes.number,
  isChild: PropTypes.bool,
  handleAddCondition: PropTypes.func,
  handleChangeCaseName: PropTypes.func,
};

Nicotine.defaultProps = {
  caseDetails: null,
  parentIndex: 0,
  isChild: false,
  handleAddCondition: () => {},
  handleChangeCaseName: () => {},
};
export default Nicotine;
