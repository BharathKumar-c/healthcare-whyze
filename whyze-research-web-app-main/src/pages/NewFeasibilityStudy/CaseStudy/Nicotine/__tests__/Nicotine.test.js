import React from 'react';
import { act, fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { setSegments } from '../../../../../redux/reducer/feasibilityStudyReducer';
import store from '../../../../../redux/store';
import Nicotine from '..';
import userEvent from '@testing-library/user-event';
import { nicotineConstants } from '../../../../../constant/ConstantTexts';

function renderWithContext(element) {
  const { container, getByText } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText };
}
const caseDetails = {
  name: 'Nicotine',
  case: 'Nicotine',
  condition: 'Inclusion',
  caseName: '',
  child: [],
};

describe('Test cases for Nicotine', () => {
  test('Create the snapshots for Nicotine', () => {
    const { container } = renderWithContext(<Nicotine />);

    act(() => {
      const { segments, activeSegment } = JSON.parse(
        JSON.stringify(store.getState().feasibilityStudyReducer),
      );
      segments[activeSegment.index].feasibilityStudy.push(caseDetails);
      store.dispatch(setSegments(segments));
    });

    expect(container).toMatchSnapshot();
  });

  test('Nicotine Condition input box', () => {
    renderWithContext(<Nicotine />);
    const inputBox = screen.getByRole('textbox');

    expect(inputBox).toBeInTheDocument();
    expect(inputBox.value).toMatch('');
    userEvent.type(inputBox, 'test12345');
    expect(inputBox.value).toBe('test12345');
  });

  test('Should render the Add icon correctly', () => {
    renderWithContext(<Nicotine />);

    const addIcon = screen.getByAltText('nicotine-add-icon');
    expect(addIcon).toBeInTheDocument();
  });

  test('Should render error text when click add icon without caseName correctly', () => {
    renderWithContext(<Nicotine caseDetails={caseDetails} />);

    const addIcon = screen.getByAltText('nicotine-add-icon');
    fireEvent.click(addIcon);

    const errorText = screen.getByText(
      nicotineConstants.nicotineConditionErrorMessage,
    );
    expect(errorText).toBeInTheDocument();
  });

  test('Should render error text when click add icon with empty caseName correctly', () => {
    const caseDetailsWithEmptyCaseName = {
      name: 'Nicotine',
      case: 'Nicotine',
      condition: 'Inclusion',
      caseName: '',
      child: [],
    };
    renderWithContext(<Nicotine caseDetails={caseDetailsWithEmptyCaseName} />);

    const addIcon = screen.getByAltText('nicotine-add-icon');
    fireEvent.click(addIcon);

    const errorText = screen.getByText(
      nicotineConstants.nicotineConditionErrorMessage,
    );
    expect(errorText).toBeInTheDocument();
  });

  test('Should render when click add icon with caseName correctly', async () => {
    const caseDetailsWithCaseName = {
      name: 'Nicotine',
      case: 'Nicotine',
      condition: 'Inclusion',
      caseName: 'Covid',
      child: [],
    };

    const handleAddCondition = jest.fn();
    renderWithContext(
      <Nicotine
        caseDetails={caseDetailsWithCaseName}
        parentIndex={2}
        handleAddCondition={handleAddCondition}
      />,
    );

    const addIcon = screen.getByAltText('nicotine-add-icon');
    fireEvent.click(addIcon);

    expect(handleAddCondition).toBeCalledTimes(1);
  });

  test('Should render vertical line when nicotine has child with condition', () => {
    const caseDetailsWithCaseName = {
      name: 'Nicotine',
      case: 'Nicotine',
      condition: 'Inclusion',
      caseName: 'Covid',
      child: [
        {
          id: 'fd8ac8ff',
          condition: 'quit',
          operation: 'And',
          child: [
            {
              id: 'e7079155',
              condition: 'started',
              operation: 'And',
              child: [],
              started: {},
            },
          ],
          quit: {
            period: 'day',
            value: '22',
          },
        },
      ],
    };
    const { container } = renderWithContext(
      <Nicotine caseDetails={caseDetailsWithCaseName} isChild />,
    );

    const verticalLine = container.getElementsByClassName('vertical-line');
    expect(verticalLine[0]).toBeInTheDocument();
  });

  test('Should render vertical line when nicotine has child without condition', () => {
    const caseDetailsWithCaseName = {
      name: 'Nicotine',
      case: 'Nicotine',
      condition: 'Inclusion',
      caseName: '',
      child: [
        {
          id: 'fd8ac8ff',
          condition: '',
          operation: 'And',
          child: [],
        },
      ],
    };
    const { container } = renderWithContext(
      <Nicotine caseDetails={caseDetailsWithCaseName} isChild />,
    );

    const verticalLine = container.getElementsByClassName(
      'vertical-line-condition',
    );
    const addButton = screen.getByText('+ Add other condition details');
    expect(verticalLine[0]).toBeInTheDocument();
    expect(addButton).toBeInTheDocument();
  });

  test('Should render vertical dashed line', () => {
    const caseDetailsWithCaseName = {
      name: 'Nicotine',
      case: 'Nicotine',
      condition: 'Inclusion',
      caseName: 'Covid',
      child: [
        {
          id: 'fd8ac8ff',
          condition: '',
          operation: 'And',
          child: [],
        },
      ],
    };
    renderWithContext(<Nicotine caseDetails={caseDetailsWithCaseName} />);
    const verticalLine = screen.getByTestId('vertical-line');

    expect(verticalLine).toHaveAttribute(
      'style',
      'border-left: 2px dashed #D3D6D9; height: 32px;',
    );
  });

  test('Should render vertical solid line', () => {
    const caseDetailsWithCaseName = {
      name: 'Nicotine',
      case: 'Nicotine',
      condition: 'Inclusion',
      caseName: '',
      child: [
        {
          id: 'fd8ac8ff',
          condition: 'quit',
          operation: 'And',
          child: [],
        },
      ],
    };
    renderWithContext(<Nicotine caseDetails={caseDetailsWithCaseName} />);
    const verticalLine = screen.getByTestId('vertical-line');

    expect(verticalLine).toHaveAttribute(
      'style',
      'border-left: 2px solid #D3D6D9; height: 32px;',
    );
  });
});
