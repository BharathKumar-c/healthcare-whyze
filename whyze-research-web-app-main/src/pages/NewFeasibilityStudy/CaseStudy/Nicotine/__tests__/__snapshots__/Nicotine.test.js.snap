// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for Nicotine Create the snapshots for Nicotine 1`] = `
<div>
  <div
    class="indication"
  >
    <div
      id="main_textbox"
    >
      <span
        class="ant-typography input-lable"
      >
        Nicotine condition
      </span>
      <input
        class="ant-input"
        data-testid="main_textbox"
        placeholder=""
        style="border-radius: 0px;"
        type="text"
        value=""
      />
    </div>
    <div
      style="height: 30px;"
    >
      <div
        class="ant-image"
        style="width: 11px; height: 11px;"
      >
        <img
          alt="nicotine-add-icon"
          class="ant-image-img"
          height="11"
          src="AddIcon.svg"
          style="height: 11px; cursor: pointer;"
          width="11"
        />
      </div>
    </div>
  </div>
</div>
`;
